src/clj/harja/kyselyt/toteumat.sql

```diff
@@ -68,26 +68,27 @@ WHERE
 
 -- name: hae-toteumien-tehtavien-summat
 -- Listaa urakan toteumien tehtävien määrien summat toimenpidekoodilla ryhmiteltynä.
-SELECT
-  toimenpidekoodi AS tpk_id,
-  SUM(tt.maara)   AS maara,
-  nimi
-FROM toteuma_tehtava tt
-  LEFT JOIN toimenpidekoodi tk
-    ON tk.id = tt.toimenpidekoodi
-  JOIN toteuma t ON tt.toteuma = t.id
-                    AND t.urakka = :urakka
-                    AND sopimus = :sopimus
-                    AND alkanut >= :alkanut
-                    AND alkanut <= :paattynyt
-                    AND tyyppi = :tyyppi :: toteumatyyppi
-                    AND tt.poistettu IS NOT TRUE
-                    AND t.poistettu IS NOT TRUE
-                    AND (:toimenpide :: INTEGER IS NULL OR tk.emo = (SELECT toimenpide
-                                                                     FROM toimenpideinstanssi
-                                                                     WHERE id = :toimenpide))
-                    AND (:tehtava :: INTEGER IS NULL OR tk.id = :tehtava)
-GROUP BY toimenpidekoodi, nimi;
+SELECT x.tpk_id, x.maara, tk.nimi
+  FROM (SELECT toimenpidekoodi AS tpk_id,
+               SUM(tt.maara)   AS maara
+          FROM toteuma_tehtava tt
+         WHERE tt.toteuma IN (SELECT id FROM toteuma t
+                               WHERE t.urakka = :urakka
+		                 AND t.sopimus = :sopimus
+			         AND t.alkanut >= :alkanut
+			         AND t.alkanut <= :paattynyt
```

Comment:
Ainahan tämä on toiminut näin, mutta en tiedä haluttaisiinko kyselyn toimivan niin, että riittää että toteuma on ollut käynnissä haetulla aikavälillä.

Käytännössähän tällaisia tilanteita ei pitäisi tulla vastaan, koska toteumanäkymässä pienin haettava aikaväli on päivän alusta päivän loppuun, eikä yli päivää kestäviä toteumia kyllä pitäisi olla.. ja jos joku toteuma ainoastaan päättyy tällä aikavälillä, nin sen sitten saa näkyviin muutammalla aikaväliä.